common/src/main/java/com/wynntils/core/managers/ManagerRegistry.java

```diff
@@ -57,105 +56,88 @@ public static void init() {
+    public static boolean removeDependency(ModelDependant dependant, Class<? extends Model> manager) {
+        if (PERSISTENT_CORE_MANAGERS.contains(manager)) {
             throw new IllegalStateException("Tried to unregister a persistent manager.");
         }
 
-        MODEL_DEPENDENCIES.putIfAbsent(dependency, new ArrayList<>());
+        List<ModelDependant> managerDependencies =
+                MODEL_DEPENDENCIES.computeIfAbsent(manager, (k) -> new ArrayList<>());
 
-        MODEL_DEPENDENCIES.get(dependency).remove(dependant);
+        managerDependencies.remove(dependant);
 
-        updateManagerState(dependency);
+        return !managerDependencies.isEmpty() || tryDisableManager(manager);
     }
 
-    public static void removeAllFeatureDependency(Feature dependant) {
-        for (Class<? extends Manager> manager : MODEL_DEPENDENCIES.keySet()) {
-            boolean removed = MODEL_DEPENDENCIES.get(manager).remove(dependant);
+    public static void removeAllFeatureDependency(ModelDependant dependant) {
+        for (Class<? extends Model> model : dependant.getModelDependencies()) {
 
-            if (removed) {
-                updateManagerState(manager);
+            List<ModelDependant> managerDependencies = MODEL_DEPENDENCIES.get(model);
+
+            if (managerDependencies != null && managerDependencies.remove(dependant)) {
+                tryDisableManager(model);
             }
         }
     }
 
-    private static void updateManagerState(Class<? extends Manager> manager) {
-        List<Feature> dependencies = MODEL_DEPENDENCIES.get(manager);
-        List<Function<?>> functionDependencies = FUNCTION_MODEL_DEPENDENCIES.get(manager);
-
-        boolean hasDependencies = (dependencies != null && !dependencies.isEmpty())
-                || (functionDependencies != null && !functionDependencies.isEmpty());
-
-        if (ENABLED_MANAGERS.contains(manager)) {
-            if (!hasDependencies) {
-                WynntilsMod.unregisterEventListener(manager);
-
-                ENABLED_MANAGERS.remove(manager);
-
-                tryDisableManager(manager);
-            }
-        } else {
-            if (hasDependencies) {
-                WynntilsMod.registerEventListener(manager);
+    private static boolean tryInitManager(Class<? extends Manager> manager) {
+        // add if not already present
+        if (!ENABLED_MANAGERS.add(manager)) return false;
 
-                ENABLED_MANAGERS.add(manager);
-
-                tryInitManager(manager);
-            }
-        }
-    }
+        WynntilsMod.registerEventListener(manager);
 
-    private static void tryInitManager(Class<? extends Manager> manager) {
         try {
             MethodUtils.invokeExactStaticMethod(manager, "init");
         } catch (IllegalAccessException | NoSuchMethodException e) {
             WynntilsMod.warn("Misconfigured init() on manager " + manager, e);
-            assert false;
```

Comment:
Why are you changing this? This was *explicitly* introduced to catch code errors with assert on.